src/Confluent.Kafka/Producer.cs

```diff
@@ -369,1113 +236,903 @@ public int Poll(int millisecondsTimeout)
-        ///     will be thrown if an attempt is made to produce a message
-        ///     and the send queue is full.
-        ///      
-        ///     Warning: blockIfQueueFull is set to true, background polling is 
-        ///     disabled and Poll is not being called in another thread, 
-        ///     this will block indefinitely.
-        /// </param>
-        /// <returns>
-        ///     A Task which will complete with the corresponding delivery report
-        ///     for this request.
-        /// </returns>
-        /// <remarks>
-        ///     If you require strict ordering of delivery reports to be maintained, 
-        ///     you should use a variant of ProduceAsync that takes an IDeliveryHandler
-        ///     parameter, not a variant that returns a Task&lt;Message&gt; because 
-        ///     Tasks are completed on arbitrary thread pool threads and can 
-        ///     be executed out of order.
-        /// </remarks>
-        public Task<Message<TKey, TValue>> ProduceAsync(string topic, TKey key, TValue val, int partition, bool blockIfQueueFull)
-            => serializingProducer.ProduceAsync(topic, key, val, partition, blockIfQueueFull);
+            var modifiedConfig = config
+                .Where(prop => 
+                    prop.Key != ConfigPropertyNames.EnableBackgroundPollPropertyName &&
+                    prop.Key != ConfigPropertyNames.EnableDeliveryReportsPropertyName &&
+                    prop.Key != ConfigPropertyNames.DeliveryReportEnabledFieldsPropertyName &&
+                    prop.Key != ConfigPropertyNames.LogDelegateName &&
+                    prop.Key != ConfigPropertyNames.ErrorDelegateName &&
+                    prop.Key != ConfigPropertyNames.StatsDelegateName);
+
+            if (modifiedConfig.Where(obj => obj.Key == "delivery.report.only.error").Count() > 0)
+            {
+                // A managed object is kept alive over the duration of the produce request. If there is no
+                // delivery report generated, there will be a memory leak. We could possibly support this 
+                // property by keeping track of delivery reports in managed code, but this seems like 
+                // more trouble than it's worth.
+                throw new ArgumentException("The 'delivery.report.only.error' property is not supported by this client");
+            }
 
-        /// <summary>
-        ///     Asynchronously send a single message to the broker.
-        ///     Refer to <see cref="ProduceAsync(string, TKey, TValue, int, bool)" /> for more information.
-        /// </summary>
-        /// <remarks>
-        ///     Blocks if the send queue is full. Warning: if background polling is disabled and Poll is
-        ///     not being called in another thread, this will block indefinitely.
-        /// </remarks>
-        public Task<Message<TKey, TValue>> ProduceAsync(string topic, TKey key, TValue val, i
```

Comment:
thats fixed now, right?

Reply:
yep. produce async will now never block (it'll throw an exception instead, delegating the waiting to the application).